ci/common.rb

```diff
@@ -178,6 +178,29 @@ def self.for(smth, max_timeout = DEFAULT_TIMEOUT)
       sh %(#{path}nosetests -s -v -A "#{nose}" #{tests_directory})
       t.reenable
     end
-    task execute: [:before_install, :install, :before_script, :script]
+
+    task :execute, :flavor do |_t, attr|
+      flavor = attr[:flavor]
+      # Check if the check uses the cache
+      cached = flavor.tasks.any? { |task| task.name == (flavor.scope.path + ':before_cache') }
```

Comment:
You're right, we should probably have a better way to do this. I think that right now `before_cache` is always executed.
Since the only issue is Windows, I added a `if Gem.win_platform?` before executing the `find` in `ci:common:before_cache`.
We should probably clean this code once again later.